modulemd/v2/meson.build

```diff
@@ -136,495 +136,78 @@ test_utils_lib = library(
-    ],
-    install : false,
-)
-test('component_rpm_v2_debug', test_v2_component_rpm,
-     env : test_env)
-test('component_rpm_v2_release', test_v2_component_rpm,
-     env : test_release_env)
-
-# -- Test Modulemd.Dependencies (C) -- #
-test_v2_dependencies = executable(
-    'dependencies_v2',
-    'tests/test-modulemd-dependencies.c',
-    dependencies : [
-        modulemd_v2_dep,
-    ],
-    link_with : [
-        test_utils_lib,
-    ],
-    install : false,
-)
-test('dependencies_v2_debug', test_v2_dependencies,
-     env : test_env)
-test('dependencies_v2_release', test_v2_dependencies,
-     env : test_release_env)
-
-# -- Test Modulemd.Module (C) -- #
-test_v2_module = executable(
-    'module_v2',
-    'tests/test-modulemd-module.c',
-    dependencies : [
-        modulemd_v2_dep,
-    ],
-    link_with : [
-        test_utils_lib,
-    ],
-    install : false,
-)
-test('module_v2_debug', test_v2_module,
-     env : test_env)
-test('module_v2_release', test_v2_module,
-     env : test_release_env)
-
-# -- Test Modulemd.ModuleIndex (C) -- #
-test_v2_module_index = executable(
-    'module_index_v2',
-    'tests/test-modulemd-moduleindex.c',
-    dependencies : [
-        modulemd_v2_dep,
-    ],
-    link_with : [
-        test_utils_lib,
-    ],
-    install : false,
-)
-test('module_index_v2_debug', test_v2_module_index,
-     env : test_env)
-test('module_index_v2_release', test_v2_module_index,
-     env : test_release_env)
-
-
-# -- Test Modulemd.ModuleIndexMerger (C) -- #
-test_v2_module_index_merger = executable(
-    'module_index_merger_v2',
-    'tests/test-modulemd-merger.c',
-    dependencies : [
-        modulemd_v2_dep,
-    ],
-    link_with : [
-        test_utils_lib,
-    ],
-    install : false,
-)
-test('module_index_merger_v2_debug', test_v2_module_index_merger,
-     env : test_env)
-test('module_index_merger_v2_release', test_v2_module_index_merger,
-     env : test_release_env)
+v2_c_tests = {
+'buildopts_v2' : [ 'tests/test-modulemd-buildopts.c' ],
+'component_module_v2'    : [ 'tests/test-modulemd-component-module.c' ],
+'component_rpm_v2'       : [ 'tests/test-modulemd-component-rpm.c' ],
+'defaults_v2'            : [ 'tests/test-modulemd-defaults.c' ],
+'defaultsv1_v2'          : [ 'tests/test-modulemd-defaults-v1.c' ],
+'dependencies_v2'        : [ 'tests/test-modulemd-dependencies.c' ],
+'module_v2'              : [ 'tests/test-modulemd-module.c' ],
+'module_index_v2'        : [ 'tests/test-modulemd-moduleindex.c' ],
+'module_index_merger_v2' : [ 'tests/test-modulemd-merger.c' ],
+'modulestream_v2'        : [ 'tests/test-modulemd-modulestream.c' ],
+'profile_v2'             : [ 'tests/test-modulemd-profile.c' ],
+'rpm_map'                : [ 'tests/test-modulemd-rpmmap.c' ],
+'service_level_v2'       : [ 'tests/test-modulemd-service-level.c' ],
+'translation_v2'         : [ 'tests/test-modulemd-translation.c' ],
+'translation_entry_v2'   : [ 'tests/test-modulemd-translation-entry.c' ],
+}
```

Comment:
Just curious... Why did you choose to now represent the source file in an array and pass that to `executable()`? It was previously just a simple string.

Reply:
The sources argument accepts either a single file or a list of files. Currently, all of the tests use only a single file, but I wanted to make it consistent with a future where some tests might require multiple files.

Reply:
I figured that was intentional, but wanted to know why. Thanks!